### PR TITLE
APT/DPKG retry logic to avoid locks

### PR DESCRIPTION
**Pull request by Darleev**
# Description
Improvement 
We need to implement retry logic for apt, to avoid possible dpkg lock issues

#### Related issue:
https://github.com/actions/virtual-environments/issues/1120

## Check list
- [+] Related issue / work item is attached
- [+] Tests are written (if applicable)

### CODE REVIEW

File: images/linux/scripts/base/apt-get-retry.sh [hunk 1]
```diff
@@ -0,0 +1,22 @@
+#!/bin/bash
```
Comment (al-cheb) [inline images/linux/scripts/base/apt-get-retry.sh:1]: Looks like duplicate the apt-retry.sh

File: images/linux/scripts/base/apt-get-retry.sh [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+i=0
+tput sc
+echo JustForTestingPurposes
```
Comment (al-cheb) [inline images/linux/scripts/base/apt-get-retry.sh:5]: Why do we need it?

File: images/linux/scripts/base/apt-get-retry.sh [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+        0 ) j="-" ;;
+        1 ) j="\\" ;;
+        2 ) j="|" ;;
+        3 ) j="/" ;;
+    esac
+    if [ -n "$TERM" ] && [ "$TERM" = "unknown" ] ; then
+        TERM=rc
+    fi
+    tput rc
+    echo -en "\r[$j] Waiting for other software managers to finish..."
```
Comment (maxim-lobanov) [inline images/linux/scripts/base/apt-get-retry.sh:17]: Why do we need this logging?

File: images/linux/scripts/base/apt-retry.sh [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+        0 ) j="-" ;;
+        1 ) j="\\" ;;
+        2 ) j="|" ;;
+        3 ) j="/" ;;
+    esac
+    if [ -n "$TERM" ] && [ "$TERM" = "unknown" ] ; then
+        TERM=rc
+    fi
+    tput rc
+    echo -en "\r[$j] Waiting for other software managers to finish..."
```
Comment (maxim-lobanov) [inline images/linux/scripts/base/apt-retry.sh:17]: Why do we need this logging?

File: images/linux/scripts/base/apt.sh [hunk 5]
```diff
@@ -18,3 +18,9 @@ apt-get install jq
 add-apt-repository -y ppa:apt-fast/stable
 apt-get update
 apt-get install apt-fast
+
+# Install retry logic for apt and apt-get
+sudo mv apt-get-retry.sh /usr/local/sbin/apt-get
```
Comment (al-cheb) [inline images/linux/scripts/base/apt.sh:23]: Why do we need to mv operation if we already copied this file?
        {
            "type": "file",
            "source": "{{template_dir}}/scripts/base/apt-get-retry.sh",
            "destination": "/usr/local/sbin/apt-get"
        },

File: images/linux/scripts/base/apt-retry.sh [hunk 6]
```diff
@@ -0,0 +1,22 @@
+#!/bin/bash
```
Comment (miketimofeev) [inline images/linux/scripts/base/apt-retry.sh:1]: `#!/bin/bash -e` or `set -e`?

File: images/linux/ubuntu1804.json [hunk 7]
```diff
@@ -54,10 +54,28 @@
...
+        {
+            "type": "file",
+            "source": "{{template_dir}}/scripts/base/apt-retry.sh",
+            "destination": "/usr/local/sbin/apt"
+        },
         {
             "type": "shell",
             "inline": [
                 "mkdir {{user `image_folder`}}",
+                "chmod -R 755 /usr/local/sbin",
```
Comment (al-cheb) [inline images/linux/ubuntu1804.json:78]: Why do we apply the second time at this line and line 60?

Comment (Darleev) [reply images/linux/ubuntu1804.json:78]: Just to return permissions back to default one from 777

File: images/linux/scripts/base/apt-retry.sh (hunk 6 above)
Comment (Darleev) [reply images/linux/scripts/base/apt-retry.sh]: thank you, I have added it

File: images/linux/scripts/base/apt-retry.sh (hunk 4 above)
Comment (Darleev) [reply images/linux/scripts/base/apt-retry.sh:16]: Just for troubleshooting purposes

File: images/linux/scripts/base/apt-get-retry.sh (hunk 1 above)
Comment (Darleev) [reply images/linux/scripts/base/apt-get-retry.sh]: the first one for apt;
the second for apt-get;

File: images/linux/scripts/base/apt-get-retry.sh (hunk 3 above)
Comment (Darleev) [reply images/linux/scripts/base/apt-get-retry.sh:16]: for troubleshooting purposes.

File: images/linux/scripts/base/apt-retry.sh (hunk 4 above)
Comment (maxim-lobanov) [reply images/linux/scripts/base/apt-retry.sh:16]: I don't think that we should bring such logging to main branch. It looks redundant and unclear. If you prefer, you can leave echo line everything else should be removed

File: images/linux/scripts/base/apt-get-retry.sh (hunk 1 above)
Comment (al-cheb) [reply images/linux/scripts/base/apt-get-retry.sh]: Why don't we use symlinks?

File: images/linux/ubuntu1804.json (hunk 7 above)
Comment (al-cheb) [reply images/linux/ubuntu1804.json:78]: Why do we need to change permissions with sudo?

**Pull request closed without merging**